tendrl/monitoring_integration/grafana/alert_dashboard.py

```diff
@@ -204,123 +201,123 @@ def create_resource_dashboard(
-                               {'message': msg})
-                else:
-                    msg = "Alert dashboard upload failed for %s-%s" % \
-                          (resource_name, new_title)
-                    logger.log("debug", NS.get("publisher_id", None),
-                               {'message': msg})
+                return resource_json
         except Exception as ex:
             logger.log("debug", NS.get("publisher_id", None),
                        {'message': str(ex)})
 
 
-def add_panel(integration_id, resource_type, resource_name, sds_name):
-    resp = None
-    alert_dashboard = alert_utils.get_alert_dashboard(
-        resource_type
-    )
-    if alert_dashboard:
+def add_panel(resources, resource_type, alert_dashboard, most_recent_panel_id):
+    for resource in resources:
+        sds_name = resource["sds_name"]
+        integration_id = resource["integration_id"]
+        resource_name = resource["resource_name"]
         try:
-            if alert_dashboard["message"] == "Dashboard not found":
-                flag = False
-            else:
-                flag = True
-        except (KeyError, AttributeError):
-            flag = True
-        if flag:
-            try:
-                if sds_name == constants.GLUSTER:
-                    # check duplicate rows
-                    if not check_duplicate(
-                        alert_dashboard,
-                        integration_id,
-                        resource_type,
-                        resource_name
-                    ):
-                        alert_row = fetch_row(alert_dashboard)
-                        add_gluster_resource_panel(
-                            alert_row,
-                            integration_id,
-                            resource_type,
-                            resource_name
-                        )
-                        dash_json = create_updated_dashboard(
-                            alert_dashboard, alert_row
-                        )
-                        resp = alert_utils.post_dashboard(dash_json)
-            except Exception as ex:
-                logger.log("error", NS.get("publisher_id", None),
-                           {'message': "Error while updating "
-                            "dashboard for %s" % resource_name})
-                raise ex
-    return resp
+            if sds_name == constants.GLUSTER:
+                alert_rows = fetch_rows(alert_dashboard)
+                add_gluster_resource_panel(
+                    alert_rows,
+                    integration_id,
+                    resource_type,
+                    resource_name,
+                    most_recent_panel_id
+                )
+                alert_dashboard = create_updated_dashboard(
+                    alert_dashboard, alert_rows
+                )
+                most_recent_panel_id = alert_rows[-1]["panels"][-1]["id"]
+        except Exception as ex:
```

Comment:
Don't we know the exact exception which would be raised in this scenario? We should capture specific ones.

Reply:
i feel here exception catch is not required, i will remove try catch from here